A basic preferences global Pane that allows the user to indicate whether they're a Power User and/or a Developer

Vinnl opened the pull request:
This adds the code by Tim that uses [the Classifier](http://solid.github.io/solid-ui/Documentation/forms-intro.html) to set Power User or Developer mode.

Fixes #144, fixes #145.

Vinnl committed:
Update preference field labels

Vinnl committed:
Use a UUID instead of a non-existent URL for form

Vinnl committed:
Re-enable the preferences Dashboard Pane

Commit:
Using subclass classifier

Vinnl committed:
Fix lint errors

Commit:
Fix duplicate rendering of contacts

timbl commented on dashboard/preferenceForm.ttl:
```diff
@@ -0,0 +1,12 @@
+:this
+<http://purl.org/dc/elements/1.1/title> "Basic preferences" ;
+a ui:Form ;
+ui:part :categorizeUser;
```
The line `ui:part :categorizeUser;` is not needed  as that information is already in the following line.   Don't know why I originally wrote both.

Vinnl replied:
Added a commit that fixes that.

timbl reviewed (changes requested):
Looks good apart from one small change

Vinnl committed:
Remove redundant field as per Tim's suggestion

megoth commented on dashboard/basicPreferences.source.ts:
```diff
@@ -1,17 +1,22 @@
 import { PaneDefinition } from '../types'
 import UI from 'solid-ui'
-import $rdf, { NamedNode } from 'rdflib'
+import * as $rdf from 'rdflib'
```
Any specific reason for why we're doing this instead of just `import { NamedNode } from 'rdflib'`? We only need that for this file.

Vinnl replied:
`$rdf.parse` is called as well, but we can still import both, of course.

megoth commented on dashboard/basicPreferences.source.ts:
```diff
@@ -1,17 +1,22 @@
   icon: UI.icons.iconBase + 'noun_Sliders_341315_000000.svg',
   name: 'basicPreferences',
-  label: () => null,
+  label: (subject) => {
```
We want to expose this as a normal pane? I thought we only wanted to expose it through the global dashboard?

Vinnl replied:
Ah, sorry, I wasn't aware that this wasn't necessary for global Panes - I thought it was commented out because it wasn't ready yet. Fixed now.

megoth commented on dashboard/basicPreferences.source.ts:
```diff
@@ -21,50 +26,88 @@ export const basicPreferencesPane: PaneDefinition = {
+    loadData(ontologyExtra, ontologyData)
+
+    async function doRender () {
+      var context = await UI.authn.logInLoadPreferences({ dom, div: container })
```
Why using `var` instead of a more specific syntax such as `const`?

Vinnl replied:
Tim uses `var` usually, and I hadn't noticed. I'll review PRs better before submitting them next time - fixed now.

megoth commented on dashboard/ontology-extra.ttl:
```diff
@@ -0,0 +1,33 @@
+  rdfs:comment """Any person who might use a Solid-based system""";
+  rdfs:subClassOf foaf:Person, schema:Person, vcard:Indivudual.
+
+solid:NewUser a rdfs:Class;
```
I think we should use these files instead of the inline versions in basicPreferences.source.ts.

Now we have inconsistent models, as solid:NewUser is commented out in basicPreferences.source.ts.

Vinnl replied:
Yes, but the functionality to do that wasn't merged yet in this branch, and I didn't want to rebase again because that caused Tim to locally delete and re-clone this repo locally the previous time. I'll delete the `.ttl` files for now, since they don't do anything currently, and we can recreate them later.

megoth reviewed (changes requested):
Just some questions and smaller things to fix - looks good ^_^

Vinnl committed:
Remove unused .ttl files

They were getting out-of-sync with the inlined versions.
In the future, we should import them directly.

Vinnl committed:
Remove useless commented-out code

Vinnl committed:
Only show basic preferences in global dashboard

Vinnl committed:
Don't use var anymore

megoth reviewed (approved):
LGTM ^_^
